fathom/src/surface/distillation.rs

```diff
@@ -799,15 +844,24 @@ impl<'interner, 'arena, 'env> Context<'interner, 'arena, 'env> {
     }
 }
 
+#[allow(clippy::type_complexity)]
 fn match_if_then_else<'arena>(
     branches: &'arena [(Const, core::Term<'arena>)],
     default_branch: Option<(Option<StringId>, &'arena core::Term<'arena>)>,
-) -> Option<(&'arena core::Term<'arena>, &'arena core::Term<'arena>)> {
+) -> Option<(
+    (Option<Option<StringId>>, &'arena core::Term<'arena>),
+    (Option<Option<StringId>>, &'arena core::Term<'arena>),
+)> {
```

Comment:
I’m… not really understanding what’s going on with the `Option<Option<StringId>>`s for the then and else branches?

Reply:
In cases of the form
```fathom
match x {
    true => ...,
    _ => ...,
}
```
or
```fathom
match x {
    false => ...,
    _ => ...,
}
```

We have to push `_` onto `local_names`, or else we get confusing bugs in distillation output. Alternatively, we could only distill matches to if-then-else syntax if they are of the form
```fathom
match x {
    true => ...,
    false => ...,
}
```